src/getting-started.md

```diff
@@ -0,0 +1,470 @@
+  involves updating git submodules and downloading a beta compiler. It doesn't
+  need to be super fast, but that can help.
+
+Building the compiler takes more than half an hour on my moderately powerful
+laptop. The first time you build the compiler, LLVM will also be built unless
+you use system LLVM (see below).
+
+Like `cargo`, the build system will use as many cores as possible. Sometimes
+this can cause you to run low on memory. You can use `-j` to adjust the number
+concurrent jobs.
+
+Also, if you don't have too much free disk space, you may want to turn off
+incremental compilation (see the "Configuring" section below). This will make
+compilation take longer, but will save a ton of space from the incremental
+caches.
+
+### Cloning
+
+You can just do a normal git clone:
+
+```shell
+git clone https://github.com/rust-lang/rust.git
+```
+
+You don't need to clone the submodules at this time. But if you want to, you
+can do the following:
+
+```sh
+# first time
+git submodule update --init --recursive
+
+# subsequent times (to pull new commits)
+git submodule update
+```
+
+**Pro tip**: if you contribute often, you may want to look at the git worktrees
+tip in [this chapter][suggested].
+
+### Configuring the Compiler
+
+The compiler has a configuration file which contains a ton of settings. We will
+provide some recommendations here that should work for most, but [check out
+this chapter for more info][config].
+
+[config]: ./building/how-to-build-and-run.html#create-a-configtoml
+
+In the top level of the repo:
+
+```shell
+cp config.toml.example config.toml
+```
+
+Then, edit `config.toml`. You may want to search for, uncomment, and update
+the following settings:
+
+- `debug = true`: enables debug symbols and `debug!` logging, takes a bit longer to compile.
+- `incremental = true`: enables incremental compilation of the compiler itself,
+  which can significantly speed things up.  This is turned off by default
+  because it's technically unsound; sometimes it will cause weird crashes.
+  Also, it can consume a lot of disk space. This has the same effect as the
+  `-i` or `--incremental` flags.
+- `llvm-config`: enables building with system LLVM. [See this chapter][sysllvm]
+  for more info. This avoids building LLVM, which can take a while.
+
+[sysllvm]: ./building/suggested.html#building-with-system-llvm
+
+### `./x.py` Intro
+
+`rustc` is a _bootstrapping_ compiler, which means that it is written in Rust. So where do you
+get the original compiler from? We use the current beta compiler
+to build the compiler. Then, we use that compiler to build itself. Thus,
+`rustc` has a 2-stage build. You can read more about bootstrapping
+[here][boot], but you don't need more to contribute.
+
+[boot]: ./building/bootstrapping.md
+
+We have a special tool `./x.py` that drives this process. It is used for
+compiling the compiler, the standard libraries, and `rustdoc`. It is also used
```

Comment:
I think I've always heard of the "standard library" as a single piece of code, but I might be wrong and this is a nitpick, really 😄
*I need those nits to make a review. Because there isn't much to say otherwise...*
```suggestion
compiling the compiler, the standard library, and `rustdoc`. It is also used
```

Reply:
Mostly I said libraries because there is std, core, alloc, proc_macro, test, etc... not sure what else to call them.